src/sentry/workflow_engine/models/detector.py

```diff
@@ -41,6 +42,11 @@ class Detector(DefaultFieldsModel, OwnerModel, JSONConfigBase):
     # If the detector is not enabled, it will not be evaluated. This is how we "snooze" a detector
     enabled = models.BooleanField(db_default=True)
 
+    # The detector's status - used for tracking deletion state
+    status = models.SmallIntegerField(
+        default=DeletionStatus.ACTIVE.value, db_default=DeletionStatus.ACTIVE.value
```

Comment:
`DeletionStatus.ACTIVE` sounds like the deletion is actively taking place rather than it's not in any state of deletion, @saponifi3d did you not want to use `ObjectStatus` (https://github.com/getsentry/sentry/blob/master/src/sentry/constants.py#L476-L482) because of the `HIDDEN` and `DISABLED` statuses we won't be using? 

What do you think about about either calling the default `DeletionStatus.INACTIVE` or renaming the enum to `WorkflowEngineObjectStatus` (I'm having a hard time coming up with another name that isn't taken)?

Reply:
I was reading `ACTIVE` as nothing is happening 😅 like the "status" of the current model is "active" - (actively monitoring).

I think this whole enum is a bit confusing if i'm honest. I'd expect something that is `INACTIVE` to not work. The way i'm thinking about this is from the detector perspective, for example: `detector.status == INACTIVE` feels like it wouldn't be reading or monitoring anything, not that it's "not being deleted".

i don't care tooo much about what fields are we're using, but i think it's important that the `detector.status` reads in an clean way. the way i had been thinking about this is: `detector.status == active` would mean it's active and can be used, not that it's in the process of being deleted.

Reply:
Ok, how about renaming the enum then? `ObjectStatus` might be confusing since we already have an enum with that name, but maybe it could be `WorkflowEngineObjectStatus` in the base typing file and then in each model file we rename it to `DetectorStatus` and `WorkflowStatus`?

Reply:
yeah - feel free to update the enum name (the `DeletionStatus` one) and have it all make better sense :) i think the naming confused me a bunch and was part of why i left this in a draft state. haha